app/src/components/common/claim_dai_modal/index.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect } from 'react'
+import styled from 'styled-components'
+
+import { State, useXdaiBridge } from '../../../hooks/useXdaiBridge'
+import theme from '../../../theme'
+import { formatBigNumber } from '../../../util/tools'
+import { Button } from '../../button/button'
+import { ButtonRound } from '../../button/button_round'
+import { ButtonType } from '../../button/button_styling_types'
+import { IconClose } from '../icons'
+import { IconArrowUp } from '../icons/IconArrowUp'
+import { DaiIcon } from '../icons/currencies'
+import { Spinner } from '../spinner'
+import { Modal, ModalBackground } from '../switch_network_modal'
+
+const ClaimAmount = styled.div`
+  margin-top: 32px;
+  font-size: 16px;
+  line-height: 18.75px;
+  color: ${props => props.theme.textfield.color};
+`
+const SecondaryText = styled.div`
+  margin-top: 12px;
+  color: ${props => props.theme.colors.textColor};
+  font-size: ${props => props.theme.fonts.defaultSize};
+  text-align: center;
+`
+const ClaimButton = styled(Button)`
+  margin-top: 32px;
+  font-weight: 500;
+`
+const CloseStyled = styled.div`
+  align-self: flex-end;
+  cursor: pointer;
+  margin-bottom: 32px;
+`
+const CloseButton = styled(ButtonRound)`
+  width: 252px;
+  margin-top: 32px;
+`
+const TransactionLink = styled.a`
+  margin-top: 16px;
+  color: ${({ theme }) => theme.colors.clickable};
+  cursor: pointer;
+`
+const Reversed = styled.div`
+  transform: rotate(180deg);
+`
+
+export const ClaimDaiModal = (props: any) => {
+  const { claimLatestToken, transactionHash, transactionStep } = useXdaiBridge()
+  //to test state just uncomment code below here and uncomment mikana in the claim button below as well as remove transactin Step for xDai bridge
+  // let mikana = 3
+  // const transactionStep =
+  //   mikana === 1
+  //     ? State.transactionSubmitted
+  //     : mikana === 2
+  //     ? State.transactionConfirmed
+  //     : mikana === 3
+  //     ? State.waitingConfirmation
+  //     : mikana === 4
+  //     ? State.error
+  //     : State.idle
+
+  useEffect(() => {
+    if (transactionStep === State.error) props.setClaim(false)
```

Comment:
Ideally we should have some better error handling. Maybe have an error component that reads the error whenever `transactionStep === State.error`

Reply:
Yeah that could be done! I proposed it as well @pimato what do you think?